### DISCUSSION TITLE
How to use java APIS customized on docker

### DISCUSSION THREAD

**Discussion by leccyril** [exadel-inc/CompreFace#814] [Q&A]
Hello,

I want to customize the java API code to add another SDK (for liveness detection and anti spoofing) and new controllers, what is the process to rebuild java all applications on docker and then use docker-compose to run all new created builds ? Just launch mvn install on parent project then launch the docker compose ? because i not see any docker file for the maven parent project.

And what is the process to work locally

Thank you !

**Comment by pospielov** [comment]
in `dev` folder there is `docker-compose` file for building the whole project and `dockerfile` for building java part of the CompreFace
Still, I would suggest not extending CompreFace for liveness detection, but creating an external logic with integration with CompreFace.
The reason is active liveness detection should work with video, but CompreFace is intended to work with images

**Reply by leccyril** [reply]
thank you, it is intended for picture but we want to minimize the number of containers used... and method to majke detection or recognition to send the expected result...

thank you for your reactivity

**Comment by leccyril** [comment]
And i have a question, why did you choose to separate all logicals in different container ? (admin , core and api?) thank you

**Reply by pospielov** [reply]
This is a good practice to separate logic into different microservices. 
It helps with development and with scalability.
